drizzlepac/haputils/deconvolve_utils.py

```diff
@@ -131,127 +144,252 @@ def fft_deconv_img(img, psf, freq_limit=0.95, sigma=None):
-                if 'photmode' in extn.header:
-                    photmode = extn.header['photmode'].lower()
-                    break
-        if 'CAL' in photmode:
-            split_str = "CAL"
-        else:
-            split_str = "MJD"
-        phot_vals = photmode.split(split_str)[0].rstrip().split(" ")
-        kw_vals = [phot_vals[0], detector]
-        filter_list = phot_vals[2:]
-
-    # Interpret filter to match best PSF
-    # crossed filters should default to widest-band filter used
-    # remove pol filters
-    for f in filter_list:
-        if f.lower().startswith('pol') or f.lower().startswith('mjd'):
-            filter_list.remove(f)
-
-    if len(filter_list) > 1:
-        bandpass = ['lp', 'w', 'm']
+    log.debug('Looking for Library PSFs:\n  {}'.format(psf_names))
+
+    psfs_exist = [os.path.exists(fname) for fname in psf_names]
+    if not all(psfs_exist):
+        log.error('Some PSF NOT found for keywords {} \n   with values of {}'.format(INSTR_KWS, filter_list))
+        if not any(psfs_exist):
+            log.error('NO PSF(s) found for keywords {} \n   with values of {}'.format(INSTR_KWS, filter_list))
+            raise ValueError
+
+    log.info("Using Library PSF(s):\n    {}".format([os.path.basename(name) for name in psf_names]))
+
+    return psf_names
+
+
+def get_filter_names(imgname):
+    """Interpret photmode from image
+
+    Parameters
+    -----------
+    imgname : str
+        Filename of observation to extract filter names from
+
+    Returns
+    --------
+    kw_vals : list
+        List containing instrument, detector, and filter names in that order.
+    """
+
+    # look for instrument, detector, and filter in input image name
+    # Start by looking in the input image header for these values, so we know what to look
+    hdu = fits.open(imgname)
+    photmode = hdu[('sci', 1)].header.get('photmode')
+    if photmode is None:
+        photmode = hdu[0].header.get('photmode')
+
+    detector = hdu[0].header.get('detector')
+    hdu.close()
+    del hdu
+
+    # Remove duplicate entries from photmode, if any (like 2 CLEAR filters)
+    filter_list = list(dict.fromkeys(photmode.split(' ')))
+    # Key off of instrument and detector (first 2 members of photmode)
+    kw_vals = [filter_list[0].lower(), detector.lower()]
+
+    # This will remove all polarizer, grism and prism filter entries as well.
+    excl_filters = ['CAL', 'MJD', 'POL', 'GR', 'PR']
+    for e in excl_filters:
+        indx = [filter_list.index(i) for i in filter_list if i.startswith(e)]
+        indx.reverse()
+        for i in indx:
+            del filter_list[i]
+
+    # Now select the widest-band filter used for this observation
+    # This accounts for cross-filter usage.
+    found_filter = False
+    if len(filter_list[2:]) >= 1:
+        # Loop over types of bandpass based on filter names
+        # going from widest band to narrowest bands
+        bandpass = ['lp', 'w', 'm', 'n']
```

Comment:
Maybe I am lost in the weeds at this point, but is this code meant to handle SBC too?  If so, SBC has a filter "BLOCK" which is for internal calibration.